src/main/java/org/walkmod/javalang/compiler/types/IndexedURLClassPath.java

```diff
@@ -0,0 +1,157 @@
+        while (lastIndexed < urls.length) {
+            indexURLs(urls[lastIndexed]);
+            lastIndexed ++;
+        }
+        return index.list(packageFile);
+    }
+
+    public List<String> listSDKContents(final String packageName) {
+        String packageFile = packageName.replaceAll("\\.", File.separator);
+        indexURLs(RT_JAR);
+        return index.list(packageFile);
+    }
+
+
+    private void indexURLs(URL url) {
+       try {
+           if (!"file".equals(url.getProtocol())) {
+               throw new RuntimeException("Classpath element is not a file: " + url);
+           }
+           File root = new File(url.getPath());
+
+           if (root.isDirectory()) {
+               String rootPath = root.getPath();
+               if (!rootPath.endsWith(File.separator)) {
+                   rootPath = rootPath + File.separator;
+               }
+               addFilesToIndex(rootPath.length(), root, url);
+           } else if (root.isFile() && root.getName().endsWith(".jar")) {
+               JarFile jarFile = new JarFile(root);
+               try {
+                   Enumeration<JarEntry> entries = jarFile.entries();
+                   while (entries.hasMoreElements()) {
+                       JarEntry entry = entries.nextElement();
+                       String name = entry.getName();
+                       maybeIndexResource(name, url);
+                   }
+               } finally {
+                   jarFile.close();
+               }
+           }
+       } catch (IOException e) {
+           throw new RuntimeException(e);
+       }
+    }
+
+    private void addFilesToIndex(int basePrefixLen, File f, URL delegate) throws IOException {
+        if (f.isDirectory()) {
+            if (f.getPath().length() > basePrefixLen) {  // Don't index the root itself.
+                String relPath = f.getPath().substring(basePrefixLen);
+                maybeIndexResource(relPath, delegate);
+            }
+            File[] directoryEntries = f.listFiles();
+
+            if (directoryEntries == null) {
+              throw new RuntimeException("The list of directories of " + f.getAbsolutePath() + " is null");
+            }
+            for (int i = 0; i < directoryEntries.length; ++i) {
+                addFilesToIndex(basePrefixLen, directoryEntries[i], delegate);
+            }
+        } else {
+            String relPath = f.getPath().substring(basePrefixLen);
+            maybeIndexResource(relPath, delegate);
+        }
+    }
+
+    public URL[] getURLs() {
+        return urls;
+    }
+
+  /**
+   * Callers may request the directory itself as a resource, and may
+   * do so with or without trailing slashes.  We do this in a while-loop
+   * in case the classpath element has multiple superfluous trailing slashes.
+   * @param relPath relative path
+   * @param delegate value to insert
+   */
+    private void maybeIndexResource(String relPath, URL delegate) {
```

Comment:
What I meant was:
```
             String cleanRelPath = stripTrailingFileSeparators();
         if (!index.containsKey(cleanRelPath)) {
￼             index.put(cleanRelPath, delegate);
￼         }
```

And using the same strip-Method for lookup method of index, too.
Seems fishy for me to add multiples here but don't strip on lookup.

Reply:
Theoretically, you could have multiple classes from different jars under the same package and the JDK uses the first that appears in the classpath.